server/src/main/java/de/pnp/manager/network/ClientHandler.java

```diff
@@ -283,6 +324,92 @@ public StringProperty clientNameProperty() {
 
         stateMachine.registerTransition(States.IN_CHARACTER, ASSIGN_CHARACTERS, new AssignCharacterHandler(this));
         stateMachine.registerTransition(States.IN_CHARACTER, States.IN_SESSION, DISMISS_CHARACTERS, new DismissCharacterHandler(this));
+        stateMachine.registerTransition(States.IN_CHARACTER, ASSIGN_INVENTORIES, new AssignInventoryHandler(this));
+        stateMachine.registerTransition(States.IN_CHARACTER, DISMISS_INVENTORIES, new DismissInventoryHandler(this));
+
+        stateMachine.registerTransition(States.IN_CHARACTER, CREATE_ITEM_REQUEST, message -> {
+            CreateItemRequestMessage.CreateItemData data = ((CreateItemRequestMessage) message).getData();
+
+            if (hasAccessToInventory(data.getInventoryID())) {
+                IInventory inventory = manager.getInventoryHandler().getInventory(getSessionID(), data.getInventoryID());
+
+                if (inventory != null && inventory.addAll(data.getItems())) {
```

Comment:
I am not sure how addAll worked in detail, but I think it could return false after e.g. half of the items were added to the inventory, so we would be in a kinda weird state here
need to check the addAll or add method again

Reply:
No. It checks first if it is able to add all items. It would return false and add no items if it cant add all items